### ISSUE TITLE
Use OAuthProxy as authentication method

### ISSUE THREAD

**Issue by grumpyoldman-io** [backstage/backstage#6296]
Hey there, in our company we use a [Oauth2 Proxy based authentication method](https://github.com/oauth2-proxy/oauth2-proxy), which when authenticated exposed a header to the underlying application with the identity of the user. I was wondering if someone could point me in the right direction to use this as an auth method.

I've tried creating a custom plugin with custom ID provider, but the client flow is very restricting in that it needs a sign-up page and pre-existing API refs (the documentation does not mention how to add a custom ApiRef in [this section](https://backstage.io/docs/auth/#adding-the-provider-to-the-sign-in-page)).

another option would be to use passport-trusted-header, but again I run in to a lack of documentation about how to add a custom provider to the client side application.

**Comment by JordanSussman** [comment]
We abused the [awsalb provider](https://github.com/backstage/backstage/blob/8b71ba6a90f71874e27e30204f28e7725b8dfa99/contrib/docs/tutorials/aws-alb-aad-oidc-auth.md) to get our internal Backstage deployment working with OAuth2 Proxy. This isn't the ideal solution since it relies on connecting to the `/oauth2/userinfo` endpoint instead of utilizing the exposed headers.

Note: you will need to update the example `DummySignInComponent` code to utilize the "fetch(`${window.location.origin}/oauth2/userinfo`)" instead of "fetch(`${window.location.origin}/api/auth/awsalb/refresh`)".

**Comment by grumpyoldman-io** [comment]
Yeah, my next step would also be to abuse an existing one, but that just feel so wrong. There is quite some documentation on how to add auth to the backend, it is just missing the parts on how to add that custom provider to the front.

**Comment by Rugvip** [comment]
@grumpyoldman-io Sorry for the delay, vacations 😅 🏝️ 

There's actually no need to add a provider to the frontend as proxy based authentication is just used for the sign-in. The only thing that's needed is a custom `SignInPage` that can deliver the user profile and Backstage token. The best reference for that is https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/aws-alb-aad-oidc-auth.md at this point
